src/tsickle_compiler_host.ts

```diff
@@ -0,0 +1,259 @@
+    const rawSourceMap = sourceMapGenerator.toJSON();
+    return new SourceMapConsumer(rawSourceMap);
+  }
+
+  sourceMapTextToConsumer(sourceMapText: string): SourceMapConsumer {
+    const sourceMapJson: any = sourceMapText;
+    return new SourceMapConsumer(sourceMapJson);
+  }
+
+  combineSourceMaps(tscSourceMapText: string): string {
+    const tscSourceMapConsumer = this.sourceMapTextToConsumer(tscSourceMapText);
+    const tscSourceMapGenerator = this.sourceMapConsumerToGenerator(tscSourceMapConsumer);
+    if (this.tsickleSourceMaps.size > 0) {
+      for (const sourceFileName of (tscSourceMapConsumer as any).sources) {
+        const tsickleSourceMapGenerator = this.tsickleSourceMaps.get(sourceFileName)!;
+        const tsickleSourceMapConsumer =
+            this.sourceMapGeneratorToConsumer(tsickleSourceMapGenerator);
+        tscSourceMapGenerator.applySourceMap(tsickleSourceMapConsumer);
+      }
+    }
+    if (this.decoratorDownlevelSourceMaps.size > 0) {
+      for (const sourceFileName of (tscSourceMapConsumer as any).sources) {
+        const decoratorDownlevelSourceMapGenerator =
+            this.decoratorDownlevelSourceMaps.get(sourceFileName)!;
+        const decoratorDownlevelSourceMapConsumer =
+            this.sourceMapGeneratorToConsumer(decoratorDownlevelSourceMapGenerator);
+        tscSourceMapGenerator.applySourceMap(decoratorDownlevelSourceMapConsumer);
+      }
+    }
+
+    return tscSourceMapGenerator.toString();
+  }
+
+  convertCommonJsToGoogModule(fileName: string, content: string): string {
+    const moduleId = this.environment.fileNameToModuleId(fileName);
+
+    let {output, referencedModules} = processES5(
+        fileName, moduleId, content, this.environment.pathToModuleName.bind(this.environment),
+        this.options.es5Mode, this.options.prelude);
+
+    const moduleName = this.environment.pathToModuleName('', fileName);
+    this.modulesManifest.addModule(fileName, moduleName);
+    for (let referenced of referencedModules) {
+      this.modulesManifest.addReferencedModule(fileName, referenced);
+    }
+
+    return output;
+  }
+
+  private runDecoratorDownlevel(
+      sourceFile: ts.SourceFile, program: ts.Program, fileName: string,
+      languageVersion: ts.ScriptTarget): ts.SourceFile {
+    if (this.environment.shouldSkipTsickleProcessing(fileName)) return sourceFile;
+    let fileContent = sourceFile.text;
+    const converted = convertDecorators(program.getTypeChecker(), sourceFile);
+    if (converted.diagnostics) {
+      this.diagnostics.push(...converted.diagnostics);
+    }
+    if (converted.output === fileContent) {
+      // No changes; reuse the existing parse.
+      return sourceFile;
+    }
+    fileContent = converted.output + this.ANNOTATION_SUPPORT;
+    this.decoratorDownlevelSourceMaps.set(fileName, converted.sourceMap);
+    return ts.createSourceFile(fileName, fileContent, languageVersion, true);
+  }
+
+  private runTsickle(
```

Comment:
`run...` sounds like you're executing something for a side effect. Maybe `transformWithTsickle(sourceFile)` is closer to what this code does? Same above.